ibu-imager/seedrestoration/seedcleanup.go

```diff
@@ -0,0 +1,162 @@
+	authFile             string
+	recertContainerImage string
+	recertSkipValidation bool
+}
+
+func NewSeedRestoration(log *logrus.Logger, ops ops.Ops, backupDir,
+	containerRegistry, authFile, recertContainerImage string, recertSkipValidation bool) *SeedRestoration {
+
+	return &SeedRestoration{
+		log:                  log,
+		ops:                  ops,
+		backupDir:            backupDir,
+		containerRegistry:    containerRegistry,
+		authFile:             authFile,
+		recertContainerImage: recertContainerImage,
+		recertSkipValidation: recertSkipValidation,
+	}
+}
+
+// CleanupSeedCluster comprises the Imager workflow for cleanup operations after creating a seed image
+// out of an SNO cluster.
+func (s *SeedRestoration) CleanupSeedCluster() error {
+	s.log.Info("Cleaning up seed cluster")
+
+	// Collect all cleanup errors to only fail fatally at the end,
+	// but still cleanup as much as possible.
+	var errors []error
+
+	s.log.Infof("Removing %s folder", s.backupDir)
+	if err := os.RemoveAll(s.backupDir); err != nil {
+		s.log.Errorf("Error removing %s: %v", s.backupDir, err)
+		errors = append(errors, err)
+	}
+
+	s.log.Infof("Removing %s folder", common.BackupChecksDir)
+	if err := os.RemoveAll(common.BackupChecksDir); err != nil {
+		s.log.Errorf("Error removing %s: %v", s.backupDir, err)
+		errors = append(errors, err)
+	}
+
+	if err := s.cleanupServiceUnits(); err != nil {
+		s.log.Errorf("Error cleaning up systemd service files: %v", err)
+		errors = append(errors, err)
+	}
+
+	if err := s.cleanupScriptFiles(); err != nil {
+		s.log.Errorf("Error cleaning up script files: %v", err)
+		errors = append(errors, err)
+	}
+
+	if s.recertSkipValidation {
+		s.log.Info("Skipping restoring crypto via recert tool")
+	} else {
+		if err := recert.RestoreOriginalSeedCrypto(s.recertContainerImage, s.authFile, s.ops); err != nil {
+			s.log.Errorf("Error restoring certificates: %v", err)
+			errors = append(errors, err)
+		}
+	}
+
+	s.log.Infof("Removing %s folder", common.OvnNodeCerts)
+	if err := os.RemoveAll(common.OvnNodeCerts); err != nil {
+		s.log.Errorf("Error removing %s: %v", s.backupDir, err)
+		errors = append(errors, err)
+	}
+
+	s.log.Infof("Removing %s folder", common.MultusCerts)
+	if err := os.RemoveAll(common.MultusCerts); err != nil {
+		s.log.Errorf("Error removing %s: %v", s.backupDir, err)
+		errors = append(errors, err)
+	}
+
+	s.log.Info("Restoring cluster services (i.e. kubelet.service unit)")
+	if _, err := s.ops.SystemctlAction("enable", "kubelet.service", "--now"); err != nil {
+		s.log.Errorf("Error enabling kubelet.service unit: %v", err)
+		errors = append(errors, err)
+	}
+
+	if len(errors) > 0 {
+		return fmt.Errorf("encountered %d errors during cleanup", len(errors))
+	}
+
+	return nil
+}
+
+func (s *SeedRestoration) cleanupServiceUnits() error {
+	dir := filepath.Join(common.InstallationConfigurationFilesDir, "services")
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
```

Comment:
Maybe we can have a util function like or something better:
FindFilesInFolderAndRun(folder string, function(fullPath string) error)  error?

Reply:
It is super ok not to do it

Reply:
Agree let's merge this first

Reply:
addressed in #122